src/sqlfluff/dialects/dialect_ansi.py

```diff
@@ -1215,12 +1218,12 @@ class TableExpressionSegment(BaseSegment):
 
     type = "table_expression"
     match_grammar = OneOf(
+        Ref("ValuesClauseSegment"),
```

Comment:
This covers
```
select *
from values(1, 2, 3);
```
It was only parsing before because it was being misinterpreted as a function